Added support to push console log to fpaste.org and validate production template

jkandasa opened the pull request:
* Added support to validate [production template](https://github.com/jaegertracing/jaeger-openshift/blob/master/production/jaeger-production-template.yml)
* Added support to publish Jenkins CI server PR job console log to the world via [Fedora pastebin](https://paste.fedoraproject.org/)

objectiser commented:
:+1: :white_check_mark: :honeybee: Jenkins CI Build`#1`: `Console log:` [external](https://paste.fedoraproject.org/paste/nR~~TiOhVjFGxj4~vxyKOQ), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/1/display/redirect)

pavolloffay commented:
What do these emojis mean? Could we use just :red_circle: for fail and :white_check_mark: for success?

I am not sure if `jq` binary should be in this repo.

jkandasa commented:
@pavolloffay 

>What do these emojis mean? Could we use just :red_circle: for fail and :white_check_mark: for success?

Sure, I will update.

> I am not sure if jq binary should be in this repo.

Please give a suggestion

objectiser commented:
:+1: :white_check_mark: :honeybee: Jenkins CI Build`#2`: `Console log:` [external](https://paste.fedoraproject.org/paste/-0tEERixZ0WyppmE7ojomg), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/2/display/redirect)

objectiser commented:
:white_check_mark: Jenkins CI Build`#3`: `Console log:` [external](https://paste.fedoraproject.org/paste/ZuKaIXQl69v5RJ08AYqq9w), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/3/display/redirect)

pavolloffay commented:
> Please give a suggestion

I think there are two options, directly install it on the machine or have a script here which would install it. I think first is better.

objectiser commented:
:white_check_mark: Jenkins CI Build`#4`: `Console log:` [external](), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/4/display/redirect)

objectiser commented:
:white_check_mark: Jenkins CI Build`#5`: `Console log:` [external](), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/5/display/redirect)

objectiser commented:
:white_check_mark: Jenkins CI Build`#6`: `Console log:` [external](https://paste.fedoraproject.org/paste/k3CUp6GxRxlzBzBQYfISng), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/6/display/redirect)

jkandasa commented:
@pavolloffay 
>I think there are two options, directly install it on the machine or have a script here which would install it. I think first is better.

I have implemented the second option with a smart way. First check `jq` file availability on `/tmp` location, if it is not available to do download on `/tmp/`. From second time download will not happen.

I am sorry to pollute with too many builds. I forget to delete old `jq` soft link from the workspace. fixed it.

objectiser commented:
:white_check_mark: Jenkins CI Build`#1`: `Console log:` [external](https://paste.fedoraproject.org/paste/kpX1Jf2Xy-B78J9qB5QaiA), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/1/display/redirect)

pavolloffay commented on jenkins-ci/push-logs.sh:
```diff
@@ -0,0 +1,53 @@
+  curl -H "Content-Type: application/json" \
+    -u $AUTHTOKEN \
+    -X POST \
+    -d "{\"body\":\"${BUILD_STATUS_ICON} Jenkins CI Build\`#${BUILD_ID}\`: \`Console log:\` [external](${FPASTE_URL}), [internal](${RUN_DISPLAY_URL})\"}" \
```
\`Console log:\` can be just Console log

jkandasa replied:
@pavolloffay removed `

pavolloffay commented on jenkins-ci/push-logs.sh:
```diff
@@ -0,0 +1,53 @@
+    ln -s /tmp/jq-linux64 jq
+  fi
+  ./jq --version
+  #consoleUrl=$(echo "http://localhost:8080/job/${JOB_NAME}/${BUILD_ID}/consoleText")
```
Remove if not needed

objectiser commented:
:white_check_mark: Jenkins CI Build`#2`: `Console log:` [external](https://paste.fedoraproject.org/paste/UEthvzY-Ks7SUTVjvp5F9w), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/2/display/redirect)

objectiser commented:
:white_check_mark: Jenkins CI Build`#3`: `Console log:` [external](https://paste.fedoraproject.org/paste/q57gZajG8-gKSt80sr9D1A), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/3/display/redirect)

objectiser commented:
:white_check_mark: Jenkins CI Build`#4`: Console log: [external](https://paste.fedoraproject.org/paste/umOgwTUGxXzWXj5XY6~wFg), [internal](https://jenkins-jaeger-ci.cloud2.jonqe.lab.eng.bos.redhat.com/job/jaegertracing/job/jaeger-openshift/job/PR-32/4/display/redirect)

pavolloffay commented:
I think it could go in. 

cc @objectiser @jpkrohling

objectiser commented:
LGTM 👍

pavolloffay commented:
@jkandasa thanks!
